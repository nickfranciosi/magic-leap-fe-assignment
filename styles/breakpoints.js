// @flow
export default {
  tablet: '768px'
};
