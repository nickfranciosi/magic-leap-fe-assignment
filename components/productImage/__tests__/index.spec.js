/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import ProductImage from '../index';
import breakpoints from '../../../styles/breakpoints';

describe('ProductImage', () => {
  test('it should render', () => {
    const tree = renderer.create(<ProductImage />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it should have min-height at mobile breakpoint', () => {
    const tree = renderer.create(<ProductImage />).toJSON();
    expect(tree).toHaveStyleRule('min-height', `250px`, {
      media: `(max-width:${breakpoints.tablet})`
    });
  });
});
