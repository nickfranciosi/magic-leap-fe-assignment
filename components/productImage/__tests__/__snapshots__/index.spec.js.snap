// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage it should render 1`] = `
.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-left: 26px;
  height: auto;
  object-fit: cover;
  background: url('') no-repeat;
  background-position: center;
  background-size: contain;
}

@media (max-width:768px) {
  .c0 {
    min-height: 250px;
    margin: 0 0 16px 0;
  }
}

<div
  className="c0"
/>
`;
